interview-questions.md

```diff
@@ -8,44 +8,65 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Your answer:
+Your answer: A gym is a Ruby frame work or Ruby specific Package manager that allows ruby apps to function
 
-Researched answer:
+Researched answer: A Ruby Gem containts a packaged Ruby application or library, they can be created and shared, they are considered libraries because they have working code within them that can allow your program to run without having to manually input the code themselves. the more common Gems are 1.Bundler which tracks and installs exact gems and versions needed for your program. 2. RSpec which is a testing framework for code. 3. Devise For user authentications and account access for websites. 4. JSON Provides an API for parsing JSON from text. 5. Nokogiri this Provides HTML, XML, SAX, and Reader parsers with XPath and CSS selector support. 6. RAILS!! which is a framework for creating web application with Ruby.
 
 3. What is a relational database? Are there other kinds of databases?
 
-Your answer:
+Your answer: A relational database allows you to pull information using action commands that relate to the information stored within the database and yes there is more than one kind of database
 
-Researched answer:
+Researched answer: Relational databases allows storage and the acess to information that relates to one another, There are three types of relations, One-to-one| many-to-many| one-to-many|
+These databases are great for displaying information in tables. There is alos Object Oriented databases, Hiearachal databases, and Network databases.
+Object Oriented databases - is a combination of relational database concepts and object-oriented principles, best when working with Object-oriented languages
+Hiearachal databases - is most suitable for use cases, There is generally a root “parent” directory of data stored as records links to other subdirectory branches. A parent record can have many child records, each can only have one parent record. Data within records are held in the form of fields, and each field can only hold one value.
+Network databases - numerous member records or files can be connected to multiple owner files and vice versa. Unlike the Hiearchal database the network database model permits each record to have multiple parent and multiple child records.
 
-4. What are primary keys? Why are they important?
 
-Your answer:
+4. What are primary keys? Why are they important? Primary keys are the main 
 
-Researched answer:
+Your answer: A primary key is alike a header in a table. For example if there is a table chart about types of pizza and the top of each column was a "primary Key" something that Identitifes each row below.
+
+Researched answer: The columns within a relational database that acts as the parent to the data under it, It is used as the identifier for the data within the table. they are important because they are used when creating, reading, updating, or deleting data in a database table. You can narrow down more specific points in the table using the Primary Keys.
 
```

Comment:
Primary keys are going to be a unique identifier for each set of data. So if we think in terms of a table , each row would have its primary key to reference.